### PR TITLE
Adding a debugger display proxy

### PR DESCRIPTION
**Pull request by lipchev**
Here's what I was thinking:

image

I haven't changed the top-level display (which defaults to the ToString()) implementation, however if we wanted, we could make this some kind of internal `ToShortString()` implementation which uses the decimal notation, when possible, switching to the "a/b" format (or possibly the "m" version) otherwise.

### CODE REVIEW

File: src/Fractions/Formatter/DecimalNotationFormatter.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ public class DecimalNotationFormatter : ICustomFormatter {
     /// <remarks>
     ///     This instance can be used to format Fraction objects into decimal string representations.
     /// </remarks>
-    public static ICustomFormatter Instance { get; } = new DecimalNotationFormatter();
```
Comment (lipchev) [inline src/Fractions/Formatter/DecimalNotationFormatter.cs:75]: We need this in order for the docs of the `DecimalNotationFormatter.Instance.Format()` method to display our custom formats (not those of the interface).

File: src/Fractions/Fraction.DebugProxy.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+using System.Globalization;
+using System.Numerics;
+using Fractions.Formatter;
+
+namespace Fractions;
+
+public readonly partial struct Fraction {
+    internal readonly struct FractionDebugView(Fraction fraction) {
+        public BigInteger A => fraction.Numerator;
+        public BigInteger B => fraction.Denominator;
```
Comment (lipchev) [inline src/Fractions/Fraction.DebugProxy.cs:11]: Was unable to make "Numerator" and "Denominator" display in the order that I wanted (in VS it's alphabetical).

File: src/Fractions/Formatter/DecimalNotationFormatter.cs (hunk 1 above)
Comment (danm-de) [reply src/Fractions/Formatter/DecimalNotationFormatter.cs:75]: Ah, right - sorry.

File: src/Fractions/Fraction.DebugProxy.cs (hunk 2 above)
Comment (danm-de) [reply src/Fractions/Fraction.DebugProxy.cs:11]: I like it.

**Pull request merged**